the-collab-lab/tcl-21-smart-shopping-list
# added next purchase date to database and listview

## Description

Added extra fields in Firebase that will record when the next purchase date for an item should be based on recent purchase date as well as frequency. 

Displays this number on the Item page.

## Related Issue

Closes #11 

## Acceptance Criteria

When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before





### After

<img src="https://media.giphy.com/media/KQgVaKZzko8nd7kQpw/giphy.gif" width="400" height="400" />




## Testing Steps / QA Criteria



From your terminal window pull down the related branch using the command git pull origin as-th-next-purchase-date and then migrate to that branch using git checkout as-th-next-purchase-date
Run npm install && npm start to launch the server
Add an item via the add item form
Navigate to the list view to see the item and it's recommended repurchase date 
You may click the purchase button to update the purchase details and change the next purchase date. Repeatedly pressing the purchase button will effect the frequency with which you are expected to buy
